EmployeeLib/Managed.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace EmployeeLib
+{
+    public class Managed: Employee
+    {
+        public Employee Manager { get; set; } = null;
+
+        public virtual void AssignManager(Employee manager)
```

Comment:
rename the parameter to employee not a manager, because it is of type employee

Reply:
I don't agree with this, the method is AssignManager so I pass the parameter named manager and this makes more sense

Reply:
the parameter should not be of type employee in the first place, it should be a manager which we did in OCP. this will cause an extra validation business logic when you try to prevent assigning a developer for example as a manager to a manager.
in addition to the extra validation it's not clean code, and does not show compilation error when the developer is writing code. it will show on run time.